src/controllers/expenses.controller.js

```diff
@@ -0,0 +1,133 @@
+'use strict';
+
+const service = require('../services/expenses.services');
+const userService = require('../services/users.services');
+
+const getExpense = (req, res) => {
+  const id = Number(req.params.id);
+  const expense = service.getById(id);
+
+  if (!id) {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  if (!expense) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  res.statusCode = 200;
+  res.send(expense);
+};
+
+const getAllExpenses = (req, res) => {
+  const {
+    userId,
+    from,
+    to,
+    categories,
+  } = req.query;
+
+  let expenses = service.getAll();
+
+  if (userId) {
+    expenses = expenses.filter(
+      expense => expense.userId === Number(userId)
+    );
+  }
+
+  if (categories) {
+    expenses = expenses.filter(
+      expense => categories.includes(expense.category)
```

Comment:
if there are no `categories` in search params we will get `undefined`
if only 1 `categories` we will get just value (....?categories=1)
And if we have many categories we will get array of values (....?categories=1&categories=2&categories=3)

![image](https://github.com/mate-academy/node_accounting-app-v2/assets/86789161/ea010564-92a2-4116-9fcf-a66739d6d01f)

You code will crash if you have only 1 category